vortex-duckdb/src/convert/array/decimal.rs

```diff
@@ -0,0 +1,276 @@
+use duckdb::core::FlatVector;
+use duckdb::vtab::arrow::WritableVector;
+use num_traits::AsPrimitive;
+use vortex_array::arrays::{BooleanBuffer, DecimalArray, DecimalValueType, NativeDecimalType};
+use vortex_array::validity::Validity;
+use vortex_array::{Array, ArrayRef};
+use vortex_buffer::Buffer;
+use vortex_dtype::{DType, DecimalDType};
+use vortex_error::{VortexResult, vortex_bail, vortex_panic};
+
+use crate::convert::array::data_chunk_adaptor::SizedFlatVector;
+use crate::convert::array::validity::write_validity_from_mask;
+use crate::{ConversionCache, FromDuckDB, FromDuckDBType, ToDuckDB};
+
+impl ToDuckDB for DecimalArray {
+    fn to_duckdb(
+        &self,
+        chunk: &mut dyn WritableVector,
+        _cache: &mut ConversionCache,
+    ) -> VortexResult<()> {
+        let mut vector = chunk.flat_vector();
+
+        // Duckdb has an assumed storage width based on the decimal width (what we call precision)
+        match (
+            precision_to_duckdb_storage_size(&self.decimal_dtype())?,
```

Comment:
Might be nice to change args order in the match such that they match the order for the generic parameters.